### PR TITLE
[3.12] gh-109125: Run mypy on `Tools/wasm` (GH-109126)

### PR DESCRIPTION
**Pull request by hugovk**
(cherry picked from commit f65497fd252a4a4df960da04d68e8316b58624c0)

~Includes the other missing ones.~

* Issue: gh-109125

### CODE REVIEW

File: .github/workflows/mypy.yml [hunk 1]
```diff
@@ -7,8 +7,12 @@ on:
       - main
   pull_request:
     paths:
-      - "Tools/clinic/**"
       - ".github/workflows/mypy.yml"
+      - "Tools/cases_generator/**"
```
Comment (corona10) [inline .github/workflows/mypy.yml]: FYI, we didn't backport the https://github.com/python/cpython/pull/108090 to the 3.12
case_generator is highly dependent on CPython version (They have quite different designs of opcodes and execution detail) so it was not proper to backport it to 3.12
As a result, case_generator will have a quite implementation different per each version.

File: .github/workflows/mypy.yml [hunk 2]
```diff
@@ -25,15 +29,24 @@ concurrency:
 
 jobs:
   mypy:
-    name: Run mypy on Tools/clinic/
+    strategy:
+      matrix:
+        target: [
+          "Tools/cases_generator",
```
Comment (corona10) [inline .github/workflows/mypy.yml]: ditto

File: .github/workflows/mypy.yml (hunk 1 above)
Comment (hugovk) [reply .github/workflows/mypy.yml]: Thanks, updated to only include the `Tools/wasm` backport.

**Comment by hugovk** [comment]
I've updated the PR only to backport the `Tools/wasm` checking, but on second thoughts, let's close this and we can add it later if needed.

**Pull request closed without merging**